### PR TITLE
:sparkles: dotnet external provider

### PR DESCRIPTION
**Pull request by djzager**

### CODE REVIEW

File: external-providers/dotnet-external-provider/examples/nerd-dinner/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+all of the necessary software to build and analyze it installed. Now all we
+need to do is:
+
+1. Start the provider.
+1. Write the provider settings file.
+1. Run the analysis
+
+# Conclusion
+
+This is awesome right?
```
Comment (pranavgaikwad) [inline external-providers/dotnet-external-provider/examples/nerd-dinner/README.md:48]: absolutely

File: external-providers/dotnet-external-provider/pkg/dotnet/service_client.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+			}
+		}
+	}
+	return provider.ProviderEvaluateResponse{
+		Matched:   len(incidents) > 0,
+		Incidents: incidents,
+	}, nil
+}
+
+func processFile(path string, regex *regexp.Regexp, positionsChan chan<- protocol.TextDocumentPositionParams, wg *sync.WaitGroup) {
```
Comment (pranavgaikwad) [inline external-providers/dotnet-external-provider/pkg/dotnet/service_client.go:143]: probably re-use MultilineGrep in provider utils?

**Review by pranavgaikwad — COMMENTED** [review]
this is a great start, mostly looks good. will wait for next round of changes

**Review by pranavgaikwad — APPROVED** [review]

File: external-providers/dotnet-external-provider/pkg/dotnet/provider.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+	// handle proxy settings
+	for k, v := range config.Proxy.ToEnvVars() {
+		os.Setenv(k, v)
+	}
+
+	codePath, err := filepath.Abs(config.Location)
+	if err != nil {
+		log.Error(err, "unable to get path to analyze")
+		panic(1)
```
Comment (pranavgaikwad) [inline external-providers/dotnet-external-provider/pkg/dotnet/provider.go:81]: any specific reason to panic and not return the error?

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
One minor conern around panic

File: external-providers/dotnet-external-provider/pkg/dotnet/provider.go (hunk 3 above)
Comment (djzager) [reply external-providers/dotnet-external-provider/pkg/dotnet/provider.go:81]: no.

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**